packages/gatsby/src/bootstrap/index.js

```diff
@@ -175,9 +175,17 @@ module.exports = async (args: BootstrapArgs) => {
 
   // Find plugins which implement gatsby-browser and gatsby-ssr and write
   // out api-runners for them.
-  const hasAPIFile = (env, plugin) =>
-    // TODO make this async...
-    glob.sync(`${plugin.resolve}/gatsby-${env}*`)[0]
+  const hasAPIFile = (env, plugin) => {
+    // The plugin loader has disabled SSR APIs for this plugin. Usually due to
+    // multiple implementations of an API that can only be implemented once
+    if (env === `ssr` && plugin.skipSSR === true) return undefined
```

Comment:
Should we disable plugins for individual APIs not all SSR APIs?

Reply:
I implemented disabling individual APIs on my checkout, but then had a thought - is there any reason for a plugin to implement `replaceRenderer` _and_ `onRenderBody`? I can't find any official or community plugins that do this.

Disabling all SSR APIs keeps things simpler as it means `api-runner-ssr.js` doesn't have to check whether it should run an API or not. I'd be tempted to roll with this implementation and then review if it causes problems.

Reply:
It does seem unlikely. If we add more APIs in the future, this could potentially become not true — that a plugin would implement multiple ssr APIs. I did a query (this one was fun `ag -l replaceRenderer | xargs -L 1 ag onRenderBody -l`) and there isn't any plugin currently that implements both. So let's go with it and see what happens. It'd be easy to change later.

Reply:
That sounds good. Nice querying :)